File: Java/UberVan.java
    import java.util.ArrayList;
    import java.util.Map;
    
    class UberVan extends Car{
        //Atributo propios de UberBlack
        //Aqui ocurre algo curioso con el tipo de dato ya que el arraylist solo permite un argumento
        //Map<String, ArrayList<String, Integer>> typeCardAccepted;
        //Y de necesitar dos quedaria con un map que contiene a otro map
        Map<String, Map<String, Integer>> typeCardAccepted;
        ArrayList<String> seatsMaterial;
        private Integer passenger;
        //Creación del metodo constructo que a su vez hereda las cosas de Car
        public UberVan(String license, Account driver/*, Map<String, Map<String, Integer>> typeCardAccepted, ArrayList<String> seatsMaterial*/){
            super(license, driver);
            /*this.typeCardAccepted = typeCardAccepted;
            this.seatsMaterial = seatsMaterial;*/
        }

        //33 el @Override nos dice que estamos sobreescribiendo este metodo/comportamiento lo cual es señal de que estamos usando polimorfismo
        @Override
        public void setPassenger(Integer passenger) {
            // TODO Auto-generated method stub
            if (passenger == 6) {
                this.passenger = passenger;
            } else {
                System.out.println("Necesitas poner 4 pasajeros");
            }
        }
    }

File: PHP/uberVan.php
<?php
require_once('car.php');
class UberVan extends Car {
    //Declaracion de los atributos propios de UberX
    public $typeCardAccepted;
    public $seatsMaterial;

    //Usamos la funcion de Car como base para pasarle los parametros propios de Uber X
    public function __construct($license, $driver, $typeCardAccepted, $seatsMaterial) {
        //Aqui en lugar de super usamos parent
        parent::__construct($license, $driver);
        //Y designamos los nuevos atributos a las variables correspondientes
        $this->typeCardAccepted = $typeCardAccepted;
        $this->seatsMaterial = $seatsMaterial;
      }

      public function printDataCar()
      {
        if ($this -> passenger != null){
          //AL parecer el "." es el + de PHP y de la forma que esta comentada
          echo "--> Licencia: {$this->license} Driver: {$this->driver->name} Tipo de tajeta aceptada {$this->typeCardAccepted} Material de los asientos {$this->seatsMaterial}";
          /*echo "Licencia:" .$this->license ."Driver: " .$this->driver->name;*/
        }
      }

      public function getPassenger(){
        return $this->passenger;
      }

      public function setPassenger($passenger){

        if ($passenger == 6) {
          $this->passenger = $passenger;
        }
        else {
          echo 'Necesitas asignar 6 pasajeros';
        }
    }
  }
?>